packages/salesforcedx-apex-replay-debugger/src/breakpoints/breakpointUtil.ts

```diff
@@ -43,28 +22,18 @@ export class BreakpointUtil {
     this.typerefMapping.clear();
 
     // set the mapping from the source line info
-    for (const info of lineBpInfo) {
-      if (!this.lineNumberMapping.has(info.uri)) {
-        this.lineNumberMapping.set(info.uri, []);
-      }
-      this.lineNumberMapping.set(info.uri, this.lineNumberMapping.get(info.uri)!.concat(info.lines));
+    lineBpInfo.map(info => {
+      this.lineNumberMapping.set(info.uri, (this.lineNumberMapping.get(info.uri) ?? []).concat(info.lines));
       this.typerefMapping.set(info.typeref, info.uri);
-    }
+    });
   }
 
-  public returnLinesForLoggingFromBreakpointArgs(bpArr: DebugProtocol.SourceBreakpoint[]): string {
-    return bpArr.map(bp => bp.line).join(',');
-  }
-
-  public getTopLevelTyperefForUri(uriInput: string): string {
-    let returnValue = '';
-    this.typerefMapping.forEach((value, key) => {
-      if (value === uriInput) {
-        if (!key.includes('$')) {
-          returnValue = key;
-        }
-      }
-    });
-    return returnValue;
+  public getTopLevelTyperefForUri(uriInput: string): string | undefined {
+    return Array.from(this.typerefMapping.entries()).find(([k, v]) => v === uriInput && !k.includes('$'))?.[0];
   }
 }
+
+export const breakpointUtil = new BreakpointUtil();
```

Comment:
you can instantiate a class and export that instance instead of doing all the singleon pattern getInstance boilerplate